pkg/kv/kvclient/kvcoord/dist_sender_test.go

```diff
@@ -4286,9 +4287,37 @@ func TestEvictionTokenCoalesce(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	sendErrors := int32(0)
-	var queriedMetaKeys sync.Map
+	// makeJoinFunc will make a function which will return once N goroutines
+	// have called it simultaneously. The first time these goroutines call
+	// this function, it will return false. After it has returned false,
+	// subsequent calls will return true.
+	const N = 2
+	makeJoinFunc := func() func() (previouslyJoined bool) {
```

Comment:
Consider the following simplification using a `WaitGroup`, which avoids the mutex and condition variable pair.

```go
makeBarrier := func(n int) func() (previouslyJoined bool) {
    var wg sync.WaitGroup
    var done atomic.Bool
    wg.Add(n)
    return func() bool {
        if done.Load() {
            return true
        }
        wg.Done()
        wg.Wait()
        done.Store(true)
        return false
    }
}
```